lib/ansible/module_utils/facts/system/distribution.py

```diff
@@ -32,6 +32,13 @@ def get_uname_version(module):
     return None
 
 
+def get_uname_release(module):
```

Comment:
Rather than creating a new function just to change the `uname` flag, let's change the method signature for `get_uname_version()` to accept flags.

```python
def get_uname_version(flags=('-v',)):
    if isinstance(flags, str):
        flags = flags.split()
    command = ['uname']
    command.extend(flags)
    return command
```

Reply:
@mattclay can you please suggest how do i mock 2 different vars with one function call (with parameters) ? Thanks!

Reply:
Here is an example of how you could do it:

```patch
diff --git a/test/units/module_utils/test_distribution_version.py b/test/units/module_utils/test_distribution_version.py
index 99dbeadc42..99295a8a8f 100644
--- a/test/units/module_utils/test_distribution_version.py
+++ b/test/units/module_utils/test_distribution_version.py
@@ -1022,11 +1022,13 @@ def test_distribution_version(am, mocker, testcase):
             data = data.strip()
         return data
 
-    def mock_get_uname_version(am):
-        return testcase.get('uname_v', None)
-
-    def mock_get_uname_release(am):
-        return testcase.get('uname_r', None)
+    def mock_get_uname(am, flags):
+        if '-v' in flags:
+            return testcase.get('uname_v', None)
+        elif '-r' in flags:
+            return testcase.get('uname_r', None)
+        else:
+            return None
 
     def mock_file_exists(fname, allow_empty=False):
         if fname not in testcase['input']:
@@ -1046,8 +1048,7 @@ def test_distribution_version(am, mocker, testcase):
         return testcase.get('platform.version', '')
 
     mocker.patch('ansible.module_utils.facts.system.distribution.get_file_content', mock_get_file_content)
-    mocker.patch('ansible.module_utils.facts.system.distribution.get_uname_version', mock_get_uname_version)
-    mocker.patch('ansible.module_utils.facts.system.distribution.get_uname_release', mock_get_uname_release)
+    mocker.patch('ansible.module_utils.facts.system.distribution.get_uname', mock_get_uname)
     mocker.patch('ansible.module_utils.facts.system.distribution._file_exists', mock_file_exists)
     mocker.patch('platform.dist', lambda: testcase['platform.dist'])
     mocker.patch('platform.system', mock_platform_system)
```